### ISSUE TITLE
nginx parameters/directives

### ISSUE THREAD

**Issue by iamnotaprogrammer** [openresty/lua-nginx-module#904]
Hello, 
Where can i find the list of parameters which can be defined or redefined in lua section?
for example:
http{
 set $backend "" ; 
rewrite_by_lua_block{
...
ngx.var.backend = address
...
}
proxy_pass $proxy_address
}
I want to know about proxy_cache_valid, proxy_ignore_client_abort, proxy_hide_header, proxy_set_header and etc...
Thanks for your answer)
sorry for my english skills

**Comment by detailyang** [comment]
Hello.

These is no complete document (I have not find too) to record the variable whether can be change or read only. I guess that maybe some variable is defined in nginx codebase and some nginx modules.

If you change a read-only ngx.var.variable, the lua vm will throw error, you can check the error log

You can checkout this [doc](https://github.com/openresty/lua-nginx-module#ngxvarvariable)

**Comment by iamnotaprogrammer** [comment]
i find only this issue which discribes some the using the varibles.
issue:
https://github.com/openresty/lua-nginx-module/issues/439

**Comment by agentzh** [comment]
@iamnotaprogrammer Right, there is no such docs. Maybe you should ask the official nginx team to at least document all their own read-only built-in nginx variables ;) And OpenResty can help with the rest by compiling a list for 3rd-party modules somewhere.
